### PR TITLE
Reactnative/themeableicons

### PR DESCRIPTION
**Pull request by raghunandhan09**
## Themeable icons/glyphs
#3053
Added changes to support the themeable icons using 'tint' property for Image type and modified Action type properties as per the requirement
## Description
 - Added 'tint' property for Image type, can be configured to theme the images as required by the user. It takes any of our foreground semantic colors (like good, default, etc)
 - Deprecated iconUrl property of Action type. It now has two properties ('url'- Image url and 'tintable'- boolean)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/5241)

### CODE REVIEW

File: source/community/reactnative/src/components/actions/action-button.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ export class ActionButton extends React.Component {
 		this.title = Constants.EmptyString;
 		this.altText = Constants.EmptyString;
 		this.type = Constants.EmptyString;
-		this.iconUrl = Constants.EmptyString;
+		this.url = Constants.EmptyString;
```
Comment (IbrahimSulai) [inline source/community/reactnative/src/components/actions/action-button.js:41]: It is iconUrl for action elements. Please refer to the schema - https://adaptivecards.io/explorer/Action.OpenUrl.html

File: source/community/reactnative/src/components/inputs/input.js [hunk 2]
```diff
@@ -233,12 +233,12 @@ export class Input extends React.Component {
 							value={this.props.value}
 						/>
 						<TouchableOpacity onPress={() => { this.onClickHandle(onExecuteAction, 'inline-action') }}>
-							{Utils.isNullOrEmpty(inlineAction.iconUrl) ?
+							{Utils.isNullOrEmpty(inlineAction.url) ?
```
Comment (IbrahimSulai) [inline source/community/reactnative/src/components/inputs/input.js:236]: It is iconUrl for action elements. Please refer to the schema - https://adaptivecards.io/explorer/Action.OpenUrl.html

File: source/community/reactnative/src/models/action-model.js [hunk 3]
```diff
@@ -6,7 +6,8 @@ export class BaseActionModel extends BaseModel{
     constructor(payload, parent) {
         super(payload, parent);
         this.title = payload.title;
-        this.iconUrl = payload.iconUrl;
+        this.url = payload.url;
```
Comment (IbrahimSulai) [inline source/community/reactnative/src/models/action-model.js:9]: It is iconUrl for action elements. Please refer to the schema - https://adaptivecards.io/explorer/Action.OpenUrl.html

File: source/community/reactnative/src/visualizer/payloads/payloads/Action.NoBody.json [hunk 4]
```diff
@@ -44,7 +44,7 @@
           "id": "input1",
           "inlineAction": {
               "type": "Action.Submit",
-              "iconUrl": "http://adaptivecards.io/content/cats/1.png",
+              "url": "http://adaptivecards.io/content/cats/1.png",
```
Comment (IbrahimSulai) [inline source/community/reactnative/src/visualizer/payloads/payloads/Action.NoBody.json:47]: It is iconUrl for action elements. Please refer to the schema - https://adaptivecards.io/explorer/Action.OpenUrl.html

File: source/community/reactnative/src/visualizer/payloads/payloads/Action.NoBody.json [hunk 5]
```diff
@@ -72,7 +72,7 @@
           "isMultiline": true,
           "inlineAction": {
               "type": "Action.Submit",
-              "iconUrl": "http://adaptivecards.io/content/cats/1.png",
+              "url": "http://adaptivecards.io/content/cats/1.png",
```
Comment (IbrahimSulai) [inline source/community/reactnative/src/visualizer/payloads/payloads/Action.NoBody.json:75]: It is iconUrl for action elements. Please refer to the schema - https://adaptivecards.io/explorer/Action.OpenUrl.html

**Review by IbrahimSulai — CHANGES_REQUESTED** [review]
Please have a look.

File: source/community/reactnative/src/components/actions/action-button.js (hunk 1 above)
Comment (raghunandhan09) [reply source/community/reactnative/src/components/actions/action-button.js:41]: Implemented as per the changes requested at https://github.com/microsoft/AdaptiveCards/issues/3053

**Review by IbrahimSulai — APPROVED** [review]
LGTM

File: source/community/reactnative/src/models/action-model.js (hunk 3 above)
Comment (vivekvijayakrishnan) [reply source/community/reactnative/src/models/action-model.js:9]: Property should be `icon'. Inside action, we can consume tintable and URL from icon object. Correct ?

Comment (raghunandhan09) [reply source/community/reactnative/src/models/action-model.js]: Yes correct, I have made the changes accordingly. Please check the same

**Review by vivekvijayakrishnan — APPROVED** [review]
LGTM

**Pull request merged**